js/flightlog_fielddefs.js

```diff
@@ -230,30 +230,35 @@ var
         "BIQUAD"
     ]),
 
-    RC_SMOOTHING_DERIVATIVE_TYPE = makeReadOnly([
-        "OFF", 
+RC_SMOOTHING_DERIVATIVE_TYPE = makeReadOnly([
         "PT1", 
         "BIQUAD"
     ]),
```

Comment:
Remove the OFF option will let old versions that used it without backward compatibility. What we do here is to have the "basic" enumeration, and modify it for some versions below, in the method of the line 467.

The same applies to the FILTER_TYPE.

Reply:
Yes however again I could not figure out the correct way to do that.  

Again, if you could show exactly what I must do for one of them, I will do for the others that are similar.  I just couldn't see an example to follow.

Reply:
There are several ways, for example, for RC_SMOOTHING_DERIVATIVE_TYPE (I do this without computer, maybe something is wrong):
1. Change the actual declaration for this:
```javascript
RC_SMOOTHING_DERIVATIVE_TYPE = [],
RC_SMOOTHING_DERIVATIVE_TYPE_COMPLETE = makeReadOnly([
    "OFF",
    "PT1",
    "BIQUAD",
]),
```
2. In the `adjustFieldDefsList` function, there are different ways to add it, for example something like:
```javascript
if (semver.gte(firmwareVersion, '4.3.0')) {
    RC_SMOOTHING_DERIVATIVE_TYPE = RC_SMOOTHING_DERIVATIVE_TYPE_COMPLETE.slice(0);
    RC_SMOOTHING_DERIVATIVE_TYPE.splice(RC_SMOOTHING_DERIVATIVE_TYPE .indexOf('OFF'), 1);
    RC_SMOOTHING_DERIVATIVE_TYPE = makeReadOnly(RC_SMOOTHING_DERIVATIVE_TYPE);
} else {
    RC_SMOOTHING_DERIVATIVE_TYPE = makeReadOnly(RC_SMOOTHING_DERIVATIVE_TYPE_COMPLETE);
}
```

Reply:
This is too difficult for me :-)  If definitely it needs to be done, I'll have to leave that 'fix' to someone with your level of understanding.
Most of the old enums can remain unchanged, I hope.
For the `FILTER_TYPE` enum, I'm hoping it's OK to remove the FIR filter, which (as far as I know) hasn't been used for many years, and added the extra PT2 and PT3 members on the end.  I hope this is OK.